docs/v1/integrations/ai21.mdx

```diff
@@ -0,0 +1,117 @@
+---
+title: AI21
+description: "Use AI21's latest models with AgentOps including:
+Jamba 1.5, Jamba Instruct, and specialized task models"
+---
+
+## AI21
+
+From [AI21's docs](https://docs.ai21.com/):
+
+AI21 provides state-of-the-art language models through a simple API, offering:
+- Multiple model sizes to balance performance and cost
+- Specialized models for specific tasks like contextual answers
+- Chat and completion endpoints
+- Enterprise-grade reliability and support
+
+AI21 supports several [models](https://docs.ai21.com/reference/models) including Jamba 1.5, Jamba Instruct, and task-specific models.
+
+## Using AgentOps with AI21
+
+### Requires `ai21>=2.0.0`
+
+AgentOps works seamlessly with AI21's Python SDK. Here's how to use it:
+
+```python
+from ai21 import AI21Client
+from ai21.models.chat import ChatMessage
+import agentops
+
+# Initialize clients
+client = AI21Client(api_key="your-api-key")
+agentops.init("your-agentops-key")
+
+# Create your messages
+messages = [
+    ChatMessage(
+        content="You are a world renowned poet in the style of Edgar Allan Poe.",
+        role="system",
+    ),
+    ChatMessage(
+        content="Write me a short poem about AI agents.",
+        role="user",
+    ),
+]
+
+# Make the API call
+response = client.chat.completions.create(
+    messages=messages,
+    model="jamba-1.5-mini",
+)
+print(response.choices[0].message.content)
+```
+
+### Streaming Support
+
+AI21 supports streaming responses:
+
+```python
+response = ""
+stream_response = client.chat.completions.create(
+    messages=messages,
+    model="jamba-instruct",
+    stream=True,
+)
+
+for chunk in stream_response:
+    response += chunk.choices[0].delta.content
```

Comment:
Need to wrap `chunk.choices[0].delta.content` in a `str(...)` because if the last chunk is None, it will error.

Reply:
Done

Reply:
Done